devito/dimension.py

```diff
@@ -28,6 +29,16 @@ def get_varname(self):
         self._count += 1
         return name
 
+    @property
+    def ccode(self):
+        """C-level variable name of this dimension"""
+        return "%s_size" % self.name if self.size is None else "%d" % self.size
```

Comment:
Yes, in principle I agree. But I think this will be best enforced through the visitor pattern that we will need for the loop engine (DLE; discussed offline). The node type then defines and enforces the relevant code components, eg. `ccode`, `cdecl` and the code generation is done via a dedicated visitor class. So I would defer this suggestion until we have that in place.